frontend/src/metabase/lib/utils.js

```diff
@@ -187,6 +194,10 @@ const MetabaseUtils = {
     const aComponents = MetabaseUtils.versionToNumericComponents(aVersion);
     const bComponents = MetabaseUtils.versionToNumericComponents(bVersion);
 
+    if (!aComponents || !bComponents) {
```

Comment:
`MetabaseUtils.versionToNumericComponents` returns an array, so I doubt it's possible to have negative aComponents or bComponents

Reply:
@uladzimirdev it can return null in case of invalid versions, such as "vUNKNOWN", that's why I added the check

Reply:
ah, is it a recursive call? I was confused by `MetabaseUtils` namespace and github which showed me https://github.com/metabase/metabase/blob/c802a2358a6967fcecd6e61d97b2e5cb6eb43c84/frontend/src/metabase/lib/utils.js#L144

Reply:
`MetabaseUtils.compareVersions` calls `MetabaseUtils.versionToNumericComponents`.

`versionToNumericComponents` returns null if it can't parse the version (the array of components if it can), `compareVersions` returns null if it's passed null or if `versionToNumericComponents` returns null for any of the inputs.

Is it clearer?

Reply:
I don't like the "namespace" too much, I have a WIP branch where i'm converting the file to typescript to avoid bugs such as this, maybe I could also remove the object and export the named functions, what do you think?